tracks/x86-64-assembly/exercises/two-fer/mentoring.md

```diff
@@ -0,0 +1,93 @@
+### Reasonable solutions
+
+Using a hand-rolled loop:
+
+```nasm
+    movzx edx, byte [rdi]
+    test dl, dl
+    je .L2
+.L1:
+    add rdi, 1
+    mov byte [rsi], dl
+    add rsi, 1
+    movzx edx, byte [rdi]
+    test dl, dl
+    jne .L1
+.L2:
+```
+
+Using the `rep movsb` instructions:
+
+```nasm
+    mov rdi, rsi
+    lea rsi, [rel symbol]
+    mov rcx, symbol_len
+    rep movsb
+```
+
+### Common errors
+
+- Forgetting to save registers with callee-save status (`rbx`, `rbp`, `r12-r15`).
+- Forgetting return. A function declaration must end with `ret`.
+
+### Common suggestions
+
+- Move the string copy code into its own function or macro to avoid code duplication.
+
+- Use registers instead of memory for saving values as they are faster.
```

Comment:
Great suggestion. Many student would like to know _why_ this is. I recommend providing at least one good article that shows this, but more is nice too.